src/functionaltests/java/com/ericsson/ei/query/QueryAggregatedObjectsTestSteps.java

```diff
@@ -94,181 +99,208 @@ public void perform_valid_query_on_newly_created_aggregated_object() throws Thro
+                .setEndpoint(entryPoint)
+                .setParam("ID", documentId)
+                .performRequest();
+
+        LOGGER.debug("Response of /queryAggregatedObject RestApi, Status Code: " + response.getStatusCodeValue() +
+                           "\nResponse: " + response.getBody().toString());
+
+        JsonNode jsonNodeResult = objMapper.readValue(response.getBody().toString(), JsonNode.class);
+        JsonNode responseEntityNode = objMapper.readValue(jsonNodeResult.get("responseEntity").asText(), JsonNode.class);
         String responseEntityFormattedString = responseEntityNode.toString().
                 substring(1, responseEntityNode.toString().length()-1);
-        
+
         LOGGER.debug("AggregatedObject from Response: " + responseEntityFormattedString);
         JsonNode responseEntityFormattedJsonNode = objMapper.readValue(responseEntityFormattedString, JsonNode.class);
         String actualTestCaseFinishedEventId = responseEntityFormattedJsonNode.get("aggregatedObject")
                 .get("testCaseExecutions").get(0).get("testCaseFinishedEventId").asText();
-        
-        assertEquals(HttpStatus.OK.toString(), Integer.toString(mvcResult.getResponse().getStatus()));
+
+        assertEquals(HttpStatus.OK.toString(), Integer.toString(response.getStatusCodeValue()));
         assertEquals("Failed to compare actual Aggregated Object TestCaseFinishedEventId:\n" + actualTestCaseFinishedEventId
                 + "\nwith expected Aggregated Object TestCaseFinishedEventId:\n" + expectedTestCaseFinishedEventId,
                 expectedTestCaseFinishedEventId, actualTestCaseFinishedEventId);
     }
-    
+
     @And("^Perform an invalid query on same Aggregated object$")
     public void perform_invalid_query_on_created_aggregated_object() throws Throwable {
         final String invalidDocumentId = "6acc3c87-75e0-4aaa-88f5-b1a5d4e6cccc";
         final String entryPoint = "/queryAggregatedObject";
         final String expectedResponse = "{\"responseEntity\":\"[]\"}";
-        
+
         LOGGER.debug("Trying an invalid query on /queryAggregatedObject RestApi with invalid documentId: " + invalidDocumentId);
-        mvcResult = mockMvc.perform(get(entryPoint)
-                .param("ID", invalidDocumentId)
-                .accept(MediaType.APPLICATION_JSON)
-                .contentType(MediaType.APPLICATION_JSON))
-                .andReturn();
-        String responseAsString = mvcResult.getResponse().getContentAsString();
-        int reponseStatusCode = mvcResult.getResponse().getStatus();
+        HttpRequest getRequest = new HttpRequest(HttpMethod.GET);
+        response = getRequest.setPort(applicationPort)
+                .setHost(hostName)
+                .setHeaders("content-type", "application/json")
+                .setHeaders("Accept", "application/json")
+                .setEndpoint(entryPoint)
+                .setParam("ID", invalidDocumentId)
```

Comment:
I guess setParam(key, value) appends to a hashmap. So method should be named putParam(key, value) or something similar. This can maybe be done in a separeate story.

Reply:
Same comment for setHeaders() -> putHeaders(key, value)